tests/test_jn.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2020 D-Wave Systems Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import nbformat
+from nbconvert.preprocessors import ExecutePreprocessor
+import unittest
+
+def run_jn(jn, timeout):
+    
+    open_jn = open(jn, "r", encoding='utf-8')
+    notebook = nbformat.read(open_jn, nbformat.current_nbformat)
+    open_jn.close()
+        
+    preprocessor = ExecutePreprocessor(timeout=timeout, kernel_name='python3')
+    preprocessor.allow_errors = True    
+    preprocessor.preprocess(notebook, {'metadata': {'path': os.path.dirname(jn)}})
+
+    errors = []
+    for cell in notebook.cells:
+        if 'outputs' in cell:
+            for output in cell['outputs']:
+                if output.output_type == 'error':
+                    if output.evalue == 'no embedding found':
+                        return notebook, ["Embedding failed"]
+                    else:
+                        errors.append(output)
```

Comment:
Since errors are contained in the `notebook` value returned, I would split this function in two:
- `run_jn: str, int -> Notebook`
- `collect_errors: Notebook -> list`

Reply:
Sure, I'll also add the `robust_run_jn()` function I use for https://github.com/dwave-examples/hybrid-computing-notebook/pull/6/files#diff-4606d4b9ef40c1d34f6fa2c0c8b2cce8 to the other two for consistency.